.github/workflows/build.yml

```diff
@@ -0,0 +1,246 @@
+name: build
+on: [push, pull_request]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  linux:
+    name: linux
+    runs-on: ubuntu-18.04
+    container: lmmsci/linux.gcc:18.04
+    env:
+      CMAKE_OPTS: -DUSE_WERROR=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_COMPILE_CACHE=ON
+      CCACHE_MAXSIZE: 500M
+      MAKEFLAGS: -j2
+    steps:
+    - name: Update and configure Git
+      run: |
+        add-apt-repository ppa:git-core/ppa
+        apt-get update
+        apt-get --yes install git
+        git config --global --add safe.directory /__w/lmms/lmms
+    - name: Check out
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        submodules: recursive
+    - name: Cache ccache data
+      uses: actions/cache@v3
+      with:
+        key: ccache-${{ github.job }}-${{ github.ref }}-${{ github.run_id }}
+        restore-keys: |
+          ccache-${{ github.job }}-${{ github.ref }}-
+          ccache-${{ github.job }}-
+        path: ~/.ccache
+    - name: Configure
+      run: |
+        source /opt/qt5*/bin/qt5*-env.sh || true
+        mkdir build && cd build
+        cmake .. $CMAKE_OPTS -DCMAKE_INSTALL_PREFIX=./install
+    - name: Build
+      run: cd build && make
```

Comment:
I know this is our defined way to build, but I've been using `cmake --build build --parallel x` lately instead. I wonder if this would allow us to combine build jobs into a yaml anchor we could then just use all over.

Reply:
Yeah, I prefer to use CMake like this too, and already used this style for the MSVC jobs. Unfortunately, `--parallel` was only introduced in CMake 3.12, so it isn't available for the Linux and MinGW jobs. I guess one could use `-- -j<x>`, or continue to use the `MAKEFLAGS` environment variable as we do right now. For now, I'll update the non-MSVC platforms to use `cmake --build build` too, and we can introduce the parallelism argument once the docker images are updated.

GitHub actions doesn't currently support YAML anchors (I tried to use them before and got an error message telling me not to), but it may in the future, and consistency never hurts.